BaseCollector.py

```diff
@@ -50,11 +51,24 @@ def logout_handles(self):
             print("Logging out from server {}".format(server))
             handle.logout()
 
+    def get_config_data(self, key=None):
+        """
+
+        :return:
+        """
+        with open(self.config) as conf:
+            content = json.load(conf)
+        if key:
+            return content[key]
```

Comment:
Interesting idea, to have this one function providing both: full and partial config, depending on how it's queried. 
Improvement could be, if you can feed in a list of keys and it returns a dict with the json-key as key and the values in there.

Reply:
Made changes to get the dictionary.